pkg/packageinstall/packageinstall.go

```diff
@@ -216,15 +273,22 @@ func (pi *PackageInstallCR) referencedPkgVersion() (datapkgingv1alpha1.Package,
 	if err != nil {
 		return datapkgingv1alpha1.Package{}, err
 	}
+	if len(pkgList.Items) == 0 {
+		return datapkgingv1alpha1.Package{}, fmt.Errorf("Package %s not found", pi.model.Spec.PackageRef.RefName)
+	}
 
 	var versionStrs []string
 	versionToPkg := map[string]datapkgingv1alpha1.Package{}
 
+	requiresClusterVersion := false
 	for _, pkg := range pkgList.Items {
 		if pkg.Spec.RefName == pi.model.Spec.PackageRef.RefName {
 			versionStrs = append(versionStrs, pkg.Spec.Version)
 			versionToPkg[pkg.Spec.Version] = pkg
 		}
+		if pkgHasK8sConstraint(&pkg) {
+			requiresClusterVersion = true
+		}
```

Comment:
this if should be within above if since we only care about a subset of packages.